library/alloc/src/vec.rs

```diff
@@ -2928,6 +2934,35 @@ impl<T> IntoIter<T> {
         self.ptr = self.buf.as_ptr();
         self.end = self.buf.as_ptr();
     }
+
+    /// Shifts the remaining elements to `offset` and then converts the whole allocation into a Vec
+    /// with `vec.len() == offset + self.len()`
+    ///
+    /// # Safety
+    ///
+    /// When a non-zero offset is passed the resulting Vec will have an uninitialized prefix
+    /// that needs to be filled before the Vec is valid again.
+    ///
+    /// * `offset + self.len()` must not exceed `self.cap`
+    /// * `offset == 0` is always valid
+    /// * `offset` must be positive
```

Comment:
Should we add `debug_assert` for these? Also, why not use `usize` for `offset` to guarantee always being positive?

Reply:
`debug_assert` in the standard library is mostly a happy lie, the asserts aren't distributed to end-users so they're never run (I don't even thing CI enables debug assertions, but I could be wrong).

Reply:
PR actions don't but try runs do. So yeah, I can add those here.

Reply:
added